dipy/align/_public.py

```diff
@@ -233,9 +233,9 @@ def write_mapping(mapping, fname):
 
     Notes
     -----
-    The data in the file is organized with shape (X, Y, Z, 2, 3, 3), such
-    that the forward mapping in each voxel is in `data[i, j, k, 0, :, :]` and
-    the backward mapping in each voxel is in `data[i, j, k, 0, :, :]`.
+    The data in the file is organized with shape (X, Y, Z, 3, 2), such
+    that the forward mapping in each voxel is in `data[i, j, k, :, 0]` and
```

Comment:
should not be `data[i, j, k, 3, 0]` instead since you define 3 above?

Reply:
I mean `3` instead of `:`

Reply:
I might be misunderstanding your question, but there are 3 components to the mapping (dx, dy, dz) so the full forward mapping is data[:, :, :, :, 0] and has dimensions `(X,Y,Z,3)`

Reply:
yes, got it, I should go to sleep... :-)